### DISCUSSION TITLE
Slow Upload and Download speed tops out at 49mB/s

### DISCUSSION THREAD

**Discussion by NobleWolf** [haugene/docker-transmission-openvpn#2105] [Q&A]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [x] I have tried using dev branch

### Config used

docker run --cap-add=NET_ADMIN -d --name=transmission4 \
-v /media/wd8tb/docker/downloads:/data \
-e OPENVPN_PROVIDER=PIA \
-e OPENVPN_CONFIG=france \
-e OPENVPN_USERNAME=xxxxxx \
-e OPENVPN_PASSWORD=xxxxxx \
-e LOCAL_NETWORK=192.168.0.0/24 \
--log-driver json-file \
--log-opt max-size=10m \
-p 9091:9091 haugene/transmission-openvpn:latest

### Current Behavior

The upload and download speed (according to transmission UI) max out 49kB/s up and down. 

### Expected Behavior

The upload and download speed should vary and not max out at 49kB/s up and down. I've seen it get over 7000kB/s up and down.

### How have you tried to solve the problem?

I've tried 
using the variable TRANSMISSION_UTP_ENABLED=false
I've always used the google dns
Tried the dev version
Tried the PIA locations: default, france, us_chicago
Created a brand new container
Issues #702 #625

Here's the results from a speediest-cli run
```
Retrieving speedtest.net configuration...
Testing from Cogent Communications (154.3.40.71)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by Bell Mobility (Toronto, ON) [12.24 km]: 142.636 ms
Testing download speed................................................................................
Download: 27.79 Mbit/s
Testing upload speed......................................................................................................
Upload: 34.78 Mbit/s
```

### Log output

Fri Dec 10 18:19:01 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10,
Fri Dec 10 18:19:01 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts,
Fri Dec 10 18:19:01 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]154.3.42.41:1198,
Fri Dec 10 18:19:01 2021 Socket Buffers: R=[212992->212992] S=[212992->212992],
Fri Dec 10 18:19:01 2021 UDP link local: (not bound),
Fri Dec 10 18:19:01 2021 UDP link remote: [AF_INET]154.3.42.41:1198,
Fri Dec 10 18:19:01 2021 TLS: Initial packet from [AF_INET]154.3.42.41:1198, sid=73fab37e c189ce7f,
Fri Dec 10 18:19:01 2021 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this,
Fri Dec 10 18:19:01 2021 VERIFY OK: depth=1, C=US, ST=CA, L=LosAngeles, O=Private Internet Access, OU=Private Internet Access, CN=Private Internet Access, name=Private Internet Access, emailAddress=secure@privateinternetaccess.com,
Fri Dec 10 18:19:01 2021 VERIFY KU OK,
Fri Dec 10 18:19:01 2021 Validating certificate extended key usage,
Fri Dec 10 18:19:01 2021 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication,
Fri Dec 10 18:19:01 2021 VERIFY EKU OK,
Fri Dec 10 18:19:01 2021 VERIFY OK: depth=0, C=US, ST=CA, L=LosAngeles, O=Private Internet Access, OU=Private Internet Access, CN=toronto423, name=toronto423,
Fri Dec 10 18:19:01 2021 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA,
Fri Dec 10 18:19:01 2021 [toronto423] Peer Connection Initiated with [AF_INET]154.3.42.41:1198,
Fri Dec 10 18:19:02 2021 SENT CONTROL [toronto423]: 'PUSH_REQUEST' (status=1),
Fri Dec 10 18:19:02 2021 PUSH: Received control message: 'PUSH_REPLY,comp-lzo no,redirect-gateway def1,route-ipv6 2000::/3,dhcp-option DNS 10.0.0.243,route-gateway 10.8.112.1,topology subnet,ping 10,ping-restart 60,ifconfig 10.8.112.163 255.255.255.0,peer-id 52,cipher AES-128-GCM',
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: timers and/or timeouts modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: compression parms modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: --ifconfig/up options modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: route options modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: route-related options modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: peer-id set,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: adjusting link_mtu to 1625,
Fri Dec 10 18:19:02 2021 OPTIONS IMPORT: data channel crypto options modified,
Fri Dec 10 18:19:02 2021 Data Channel: using negotiated cipher 'AES-128-GCM',
Fri Dec 10 18:19:02 2021 Outgoing Data Channel: Cipher 'AES-128-GCM' initialized with 128 bit key,
Fri Dec 10 18:19:02 2021 Incoming Data Channel: Cipher 'AES-128-GCM' initialized with 128 bit key,
Fri Dec 10 18:19:02 2021 ROUTE_GATEWAY 172.17.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:11:00:05,
Fri Dec 10 18:19:02 2021 GDG6: remote_host_ipv6=n/a,
Fri Dec 10 18:19:02 2021 ROUTE6: default_gateway=UNDEF,
Fri Dec 10 18:19:02 2021 OpenVPN ROUTE6: OpenVPN needs a gateway parameter for a --route-ipv6 option and no default was specified by either --route-ipv6-gateway or --ifconfig-ipv6 options,
Fri Dec 10 18:19:02 2021 OpenVPN ROUTE: failed to parse/resolve route for host/network: 2000::/3,
Fri Dec 10 18:19:02 2021 TUN/TAP device tun0 opened,
Fri Dec 10 18:19:02 2021 TUN/TAP TX queue length set to 100,
Fri Dec 10 18:19:02 2021 /sbin/ip link set dev tun0 up mtu 1500,
Fri Dec 10 18:19:02 2021 /sbin/ip addr add dev tun0 10.8.112.163/24 broadcast 10.8.112.255,
Fri Dec 10 18:19:02 2021 /sbin/ip route add 154.3.42.41/32 via 172.17.0.1,
Fri Dec 10 18:19:02 2021 /sbin/ip route add 0.0.0.0/1 via 10.8.112.1,
Fri Dec 10 18:19:02 2021 /sbin/ip route add 128.0.0.0/1 via 10.8.112.1,
Fri Dec 10 18:19:02 2021 WARNING: OpenVPN was configured to add an IPv6 route over tun0. However, no IPv6 has been configured for this interface, therefore the route installation may fail or may not work as expected.,
Up script executed with device=tun0 ifconfig_local=10.8.112.163,
Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.8.112.163,
Using Combustion UI, overriding TRANSMISSION_WEB_HOME,
,
-------------------------------------,
Transmission will run as,
-------------------------------------,
User name:   root,
User uid:    0,
User gid:    0,
-------------------------------------,
,
Updating Transmission settings.json with values from env variables,
Using existing settings.json for Transmission /data/transmission-home/settings.json,
Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.8.112.163,
Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed,
Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete,
Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091,
Overriding utp-enabled because TRANSMISSION_UTP_ENABLED is set to false,
Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch,
sed'ing True to true,
STARTING TRANSMISSION,
Provider PIA has a script for automatic port forwarding. Will run it now.,
If you want to disable this, set environment variable DISABLE_PORT_UPDATER=true,
Transmission startup script complete.,
Fri Dec 10 18:19:02 2021 Initialization Sequence Completed,
Running functions for token based port fowarding,
Reserved Port: 59606  Fri Dec 10 18:19:08 UTC 2021,
transmission auth not required,
waiting for transmission to become responsive,
transmission became responsive,
  50   100%    1.21 GB  Done         0.0     0.0    0.0  Will Verify  Fixer.Upper.S02E07.A.College.Chaplain.Seeks.Home.to.Keep.the.Peace.720p.HDTV.x264-W4F[eztv].mkv,
Sum:          44.58 GB               0.0     0.0,
setting transmission port to 59606,
localhost:9091/transmission/rpc/ responded: "success",
Checking port...,
Port is open: Yes,
#######################,
        SUCCESS        ,
#######################,
Port: 59606,
Expiration Fri Feb 11 06:15:07 UTC 2022,
#######################,
Entering infinite while loop,
Every 15 minutes, check port status,
Fri Dec 10 18:31:14 2021 event_wait : Interrupted system call (code=4),
Fri Dec 10 18:31:14 2021 /etc/openvpn/tunnelDown.sh tun0 1500 1553 10.8.112.163 255.255.255.0 init,
resolv.conf was restored,
Sending kill signal to transmission-daemon,
Waiting 5s for transmission-daemon to die,
Successfuly closed transmission-daemon,
Fri Dec 10 18:31:18 2021 /sbin/ip route del 154.3.42.41/32,
Fri Dec 10 18:31:18 2021 /sbin/ip route del 0.0.0.0/1,
Fri Dec 10 18:31:18 2021 /sbin/ip route del 128.0.0.0/1,
Fri Dec 10 18:31:18 2021 Closing TUN/TAP interface,
Fri Dec 10 18:31:18 2021 /sbin/ip addr del dev tun0 10.8.112.163/24,
Fri Dec 10 18:31:18 2021 SIGTERM[hard,] received, process exiting,

### Environment

```markdown
- OS: Ubuntu 20.04.2 LTS
- Docker: version 20.10.6, build 370c289
```

### Anything else?

The up/down speed does not change if I have 1 or many torrents running. 
My internet speed is 100 up and 50 down. 
I've tried using the PIA app on another device on the same network and get great speeds. 
I've attempted downloading on another machine using deluge and the same PIA VPN and get good speeds.
I've done a speed test on the host machine without VPN internet speed is good as well.
The host machine is idling at normal CPU and RAM usage. ~10% CPU and 4.5gb/15.4 gb ram.

**Comment by edgd1er** [comment]
You limitation may be due to your vpn provider and/or the server load. Have you test the speed from within the container ?

```
docker exec <yourcontainername> bash
pip install --user speedtest-cli
speedtest-cli
```

**Comment by pkishino** [comment]
This isn’t really a container bug or issue,
Feel free to run a Speedtest inside the container as mentioned above.
Try different torrents from different trackers.. it could be tracker limiting etc etc, nothing container related

**Comment by NobleWolf** [comment]
@edgd1er I did a speed test in that container. Check the "How have you tried to solve the problem?" in my original post for the results.

@pkishino I attempted different trackers and different torrents. I also ran deluge with the torrents on a different machine with the same VPN location and got my normal 7+ MB/s speed.

**Comment by katekyonni** [comment]
In the past I also had the same problem and very slow speeds, I solved this by using the openvpn-tcp config files istead of the default openvpn which are udp.
I did that by adding the environment variable `PIA_OPENVPN_CONFIG_BUNDLE=openvpn-tcp`. I found the available files and the variable here https://github.com/haugene/docker-transmission-openvpn/blob/master/openvpn/pia/configure-openvpn.sh

I don't know if that is the correct way to do it, and it could break things in the future, but it's working for me 😃 
I also have no idea why there is such a speed difference between tcp and udp for me. Maybe it's the same for you 🤷

**Reply by NobleWolf** [reply]
I tried adding the environment variable, but it was no change for me. :(

**Comment by NobleWolf** [comment]
Anyone have any other suggestions?
Using the same VPN on another machine shows good speeds. I've tried changing ports and doing as @katekyonni suggested with the TCP environment variable. But so far nothing has helped.

**Comment by aterriblegamer** [comment]
I am seeing the same behavior. Capping at 47 kB/s no matter what I have tried. 

Running image on my Synology in Docker. Has been working fine for a very long time. 

Grabbed the latest image and now it s barely moving.

Updated from my Custom ovpn to a default config for my provider. No Change. 

Speedtest from inside the container
```
Testing from ********* ...                                                                                                              
Retrieving speedtest.net server list...                                                                                                                 
Selecting best server based on ping...                                                                                                                  
Hosted by *********  [0.31 km]: 10.834 ms                                                                                            
Testing download speed................................................................................                                                  
Download: 86.76 Mbit/s                                                                                                                                  
Testing upload speed......................................................................................................                              
Upload: 177.31 Mbit/s
```

**Comment by nadav74** [comment]
I can confirm having a similar issue, myself using Nordvpn. Using TCP instead of UDP is a little less horribly slow than UDP, but still nothing similar to the way it used to be until I "docker-compose pull"-ed a while ago. This leads me to speculate that this is an image issue somehow.

docker-compose.yml:
```
version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - '/somewhere_valid:/data'
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - OPENVPN_CONFIG=chXXX.nordvpn.com
            - OPENVPN_USERNAME=xxx
            - OPENVPN_PASSWORD=xxx
            - NORDVPN_PROTOCOL=TCP // <- BUT it doesn't matter, may be ommitted too
            - LOCAL_NETWORK=192.168.2.0/24
            - TZ=...
            - PUID=1000
            - PGID=1000
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn
        restart: unless-stopped
```

log:
```
transmission-openvpn_1  | STARTING TRANSMISSION
transmission-openvpn_1  | Transmission startup script complete.
transmission-openvpn_1  | Mon Feb  7 21:26:24 2022 /sbin/ip route add 195.216.219.149/32 via 172.18.0.1
transmission-openvpn_1  | Mon Feb  7 21:26:24 2022 /sbin/ip route add 0.0.0.0/1 via 10.7.2.1
transmission-openvpn_1  | Mon Feb  7 21:26:24 2022 /sbin/ip route add 128.0.0.0/1 via 10.7.2.1
transmission-openvpn_1  | Mon Feb  7 21:26:24 2022 Initialization Sequence Completed
transmission-openvpn_1  | Mon Feb  7 21:27:10 2022 event_wait : Interrupted system call (code=4)
transmission-openvpn_1  | Mon Feb  7 21:27:10 2022 /etc/openvpn/tunnelDown.sh tun0 1500 1587 10.7.2.6 255.255.255.0 init
transmission-openvpn_1  | resolv.conf was restored
transmission-openvpn_1  | Sending kill signal to transmission-daemon
transmission-openvpn_1  | Waiting 5s for transmission-daemon to die
transmission-openvpn_1  | Successfuly closed transmission-daemon
transmission-openvpn_1  | Mon Feb  7 21:27:13 2022 /sbin/ip route del 195.216.219.149/32
transmission-openvpn_1  | Mon Feb  7 21:27:13 2022 /sbin/ip route del 0.0.0.0/1
transmission-openvpn_1  | Mon Feb  7 21:27:13 2022 /sbin/ip route del 128.0.0.0/1
transmission-openvpn_1  | Mon Feb  7 21:27:13 2022 Closing TUN/TAP interface
transmission-openvpn_1  | Mon Feb  7 21:27:13 2022 /sbin/ip addr del dev tun0 10.7.2.6/24
transmission-openvpn_1  | Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
transmission-openvpn_1  | Creating TUN device /dev/net/tun
transmission-openvpn_1  | Using OpenVPN provider: NORDVPN
transmission-openvpn_1  | Running with VPN_CONFIG_SOURCE auto
transmission-openvpn_1  | Provider NORDVPN has a bundled setup script. Defaulting to internal config
transmission-openvpn_1  | Executing setup script for NORDVPN
transmission-openvpn_1  | Downloading user specified config. NORDVPN_PROTOCOL is set to: TCP
transmission-openvpn_1  | 2022-02-07 21:27:23 Checking curl installation
transmission-openvpn_1  | 2022-02-07 21:27:23 Removing existing configs
transmission-openvpn_1  | 2022-02-07 21:27:23 Selecting the best server...
transmission-openvpn_1  | 2022-02-07 21:27:23 Searching for group: legacy_p2p
transmission-openvpn_1  | 2022-02-07 21:27:23 Searching for technology: openvpn_tcp
transmission-openvpn_1  | 2022-02-07 21:27:23 Best server : xxxxx.nordvpn.com
transmission-openvpn_1  | 2022-02-07 21:27:23 Downloading config: default.ovpn
transmission-openvpn_1  | 2022-02-07 21:27:23 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/de1081.nordvpn.com.tcp.ovpn
transmission-openvpn_1  | 2022-02-07 21:27:23 Using OpenVPN CONFIG :: ch330.nordvpn.com
transmission-openvpn_1  | 2022-02-07 21:27:23 Downloading config: ch330.nordvpn.com.ovpn
transmission-openvpn_1  | 2022-02-07 21:27:23 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/ch330.nordvpn.com.tcp.ovpn
transmission-openvpn_1  | Starting OpenVPN using config ch330.nordvpn.com.ovpn
transmission-openvpn_1  | Modifying /etc/openvpn/nordvpn/ch330.nordvpn.com.ovpn for best behaviour in this container
transmission-openvpn_1  | Modification: Point auth-user-pass option to the username/password file
transmission-openvpn_1  | Modification: Change ca certificate path
transmission-openvpn_1  | Modification: Change ping options
transmission-openvpn_1  | Modification: Update/set resolv-retry to 15 seconds
transmission-openvpn_1  | Modification: Change tls-crypt keyfile path
transmission-openvpn_1  | Modification: Set output verbosity to 3
transmission-openvpn_1  | Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
transmission-openvpn_1  | Setting OpenVPN credentials...
transmission-openvpn_1  | adding route to local network 192.168.2.0/24 via 172.18.0.1 dev eth0
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 NOTE: --fast-io is disabled since we are not using UDP
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 TCP/UDP: Preserving recently used remote address: [AF_INET]195.216.219.149:443
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 Socket Buffers: R=[131072->131072] S=[16384->16384]
transmission-openvpn_1  | Mon Feb  7 21:27:24 2022 Attempting to establish TCP connection with [AF_INET]195.216.219.149:443 [nonblock]
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 TCP connection established with [AF_INET]195.216.219.149:443
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 TCP_CLIENT link local: (not bound)
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 TCP_CLIENT link remote: [AF_INET]195.216.219.149:443
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 TLS: Initial packet from [AF_INET]195.216.219.149:443, sid=cf07525b 7d2b31a8
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA6
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 VERIFY KU OK
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 Validating certificate extended key usage
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 VERIFY EKU OK
transmission-openvpn_1  | Mon Feb  7 21:27:25 2022 VERIFY OK: depth=0, CN=ch330.nordvpn.com
transmission-openvpn_1  | Mon Feb  7 21:27:26 2022 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 4096 bit RSA
transmission-openvpn_1  | Mon Feb  7 21:27:26 2022 [ch330.nordvpn.com] Peer Connection Initiated with [AF_INET]195.216.219.149:443
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 SENT CONTROL [ch330.nordvpn.com]: 'PUSH_REQUEST' (status=1)
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,sndbuf 524288,rcvbuf 524288,explicit-exit-notify,comp-lzo no,route-gateway 10.7.3.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.7.3.5 255.255.255.0,peer-id 0,cipher AES-256-GCM'
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: timers and/or timeouts modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: --explicit-exit-notify can only be used with --proto udp
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: compression parms modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: --sndbuf/--rcvbuf options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 Socket Buffers: R=[131072->425984] S=[87040->425984]
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: --ifconfig/up options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: route options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: route-related options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: peer-id set
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: adjusting link_mtu to 1659
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 OPTIONS IMPORT: data channel crypto options modified
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 Data Channel: using negotiated cipher 'AES-256-GCM'
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 ROUTE_GATEWAY 172.18.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:12:00:02
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 TUN/TAP device tun0 opened
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 TUN/TAP TX queue length set to 100
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 /sbin/ip link set dev tun0 up mtu 1500
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 /sbin/ip addr add dev tun0 10.7.3.5/24 broadcast 10.7.3.255
transmission-openvpn_1  | Mon Feb  7 21:27:27 2022 /etc/openvpn/tunnelUp.sh tun0 1500 1587 10.7.3.5 255.255.255.0 init
transmission-openvpn_1  | Up script executed with tun0 1500 1587 10.7.3.5 255.255.255.0 init
transmission-openvpn_1  | Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.7.3.5
transmission-openvpn_1  | Updating Transmission settings.json with values from env variables
transmission-openvpn_1  | Using existing settings.json for Transmission /data/transmission-home/settings.json
transmission-openvpn_1  | Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.7.3.5
transmission-openvpn_1  | Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed
transmission-openvpn_1  | Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete
transmission-openvpn_1  | Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
transmission-openvpn_1  | Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch
transmission-openvpn_1  | sed'ing True to true
transmission-openvpn_1  | Enforcing ownership on transmission config directories
transmission-openvpn_1  | Applying permissions to transmission config directories
transmission-openvpn_1  | Setting owner for transmission paths to 1000:1000
transmission-openvpn_1  | Setting permissions for download and incomplete directories
transmission-openvpn_1  | Mask: 002
transmission-openvpn_1  | Directories: 775
transmission-openvpn_1  | Files: 664
transmission-openvpn_1  | Setting permission for watch directory (775) and its files (664)
transmission-openvpn_1  | 
transmission-openvpn_1  | -------------------------------------
transmission-openvpn_1  | Transmission will run as
transmission-openvpn_1  | -------------------------------------
transmission-openvpn_1  | User name:   abc
transmission-openvpn_1  | User uid:    1000
transmission-openvpn_1  | User gid:    1000
transmission-openvpn_1  | -------------------------------------
transmission-openvpn_1  | 
transmission-openvpn_1  | STARTING TRANSMISSION
transmission-openvpn_1  | Transmission startup script complete.
transmission-openvpn_1  | Mon Feb  7 21:27:28 2022 /sbin/ip route add 195.216.219.149/32 via 172.18.0.1
transmission-openvpn_1  | Mon Feb  7 21:27:28 2022 /sbin/ip route add 0.0.0.0/1 via 10.7.3.1
transmission-openvpn_1  | Mon Feb  7 21:27:28 2022 /sbin/ip route add 128.0.0.0/1 via 10.7.3.1
transmission-openvpn_1  | Mon Feb  7 21:27:28 2022 Initialization Sequence Completed
```

**Comment by nadav74** [comment]
I don't believe it, but this is some sort of a UI issue. Somehow the Alternative Speed Limits are always used even without being scheduled. Just set them to high values and this issue is gone...

![Selection_862](https://user-images.githubusercontent.com/47421640/154798940-9d993329-47dd-49d4-ab29-b21799a14d75.png)
:

**Reply by pkishino** [reply]
that looks like something has gone wrong with your transmission config then.
Have you checked that you havent activated the alternativ speed toggle?
Cause that definitely isn't a problem for me at least

**Reply by nadav74** [reply]
This is not something I'd ever activate willingly. The screenshot also shows that it's supposed to be inactive.

**Reply by pkishino** [reply]
No, the screenshot shows the settings. There is a separate toggle in the main UI to toggle alternative speed. Looks like a turtle.
You could also post your transmission settings.json file to verify

**Reply by nadav74** [reply]
You just convinced me even further that I was right about it.
```
{
    "alt-speed-down": 100000,
    "alt-speed-enabled": false,
    "alt-speed-time-begin": 540,
    "alt-speed-time-day": 127,
    "alt-speed-time-enabled": false,
    "alt-speed-time-end": 1020,
    "alt-speed-up": 100000,
    "bind-address-ipv4": "10.7.0.2",
    "bind-address-ipv6": "::",
    "blocklist-enabled": false,
    "blocklist-url": "http://www.example.com/blocklist",
    "cache-size-mb": 4,
    "dht-enabled": true,
    "download-dir": "/data/completed",
    "download-limit": 100,
    "download-limit-enabled": 0,
    "download-queue-enabled": true,
    "download-queue-size": 5,
    "encryption": 1,
    "idle-seeding-limit": 30,
    "idle-seeding-limit-enabled": false,
    "incomplete-dir": "/data/incomplete",
    "incomplete-dir-enabled": true,
    "lpd-enabled": false,
    "max-peers-global": 200,
    "message-level": 2,
    "peer-congestion-algorithm": "",
    "peer-id-ttl-hours": 6,
    "peer-limit-global": 400,
    "peer-limit-per-torrent": 100,
    "peer-port": 51413,
    "peer-port-random-high": 65535,
    "peer-port-random-low": 49152,
    "peer-port-random-on-start": false,
    "peer-socket-tos": "default",
    "pex-enabled": true,
    "port-forwarding-enabled": false,
    "preallocation": 1,
    "prefetch-enabled": true,
    "queue-stalled-enabled": true,
    "queue-stalled-minutes": 30,
    "ratio-limit": 2,
    "ratio-limit-enabled": false,
    "rename-partial-files": true,
    "rpc-authentication-required": false,
    "rpc-bind-address": "0.0.0.0",
    "rpc-enabled": true,
    "rpc-host-whitelist": "",
    "rpc-host-whitelist-enabled": false,
    "rpc-password": "{8f9f90669ae8cf503a9c997f22611d2fa82b0995JO/2nQC7",
    "rpc-port": 9091,
    "rpc-url": "/transmission/",
    "rpc-username": "username",
    "rpc-whitelist": "127.0.0.1",
    "rpc-whitelist-enabled": false,
    "scrape-paused-torrents-enabled": true,
    "script-torrent-done-enabled": false,
    "script-torrent-done-filename": "",
    "seed-queue-enabled": false,
    "seed-queue-size": 10,
    "speed-limit-down": 100,
    "speed-limit-down-enabled": false,
    "speed-limit-up": 100,
    "speed-limit-up-enabled": true,
    "start-added-torrents": true,
    "trash-original-torrent-files": false,
    "umask": 2,
    "upload-slots-per-torrent": 14,
    "utp-enabled": true,
    "watch-dir": "/data/watch",
    "watch-dir-enabled": true,
    "watch-dir-force-generic": false
}
```

**Comment by pkishino** [comment]
Hmm,
Could you try and stop container,
Remove the following from settings:


"alt-speed-down": 100000,
    "alt-speed-enabled": false,
    "alt-speed-time-begin": 540,
    "alt-speed-time-day": 127,
    "alt-speed-time-enabled": false,
    "alt-speed-time-end": 1020,
    "alt-speed-up": 100000,


And then start again and check if same problem?

I know this was a bug that was supposed to have been fixed in older
transmission releases

**Reply by jorlandobr** [reply]
As @pkishino said, check settings.json.

I have been getting very slow download speeds for a few weeks, even reinstalled transmission. Just as @ndav74 suggested, I set Alternative Speeds to a higher value and then the download speeds came back to usual.

Checked the settings.json, and found it:

```
root@OMV:/srv/mergerfs/HDs_2TB/torrents/transmission-home# more settings.json
{
    "alt-speed-down": 50,
    "alt-speed-enabled": true,
    "alt-speed-time-begin": 540,
    "alt-speed-time-day": 127,
    "alt-speed-time-enabled": false,
    "alt-speed-time-end": 1020,
    "alt-speed-up": 50,
    ...
```
    
   Deleted it and redployed the container, checked again, alt-speed-enabled,was false:
   
```
root@OMV:/srv/mergerfs/HDs_2TB/torrents/transmission-home# more settings.json
{
    "alt-speed-down": 500000,
    "alt-speed-enabled": false,
    "alt-speed-time-begin": 540,
    "alt-speed-time-day": 127,
    "alt-speed-time-enabled": false,
    "alt-speed-time-end": 1020,
    "alt-speed-up": 50,
    ...
```
   
   My guess: accidently pressed the turtle, the option was cached by the browser, even after a new install the problem persisted.
